src/register.jl

```diff
@@ -0,0 +1,138 @@
+greaterthan05(x) = x .> 0.5 # used for the image resize step and for binarizing images
+function imrotate_bin(x, r)
+    return greaterthan05(collect(imrotate(x, r, axes(x); method=BSpline(Constant()))))
+end
+function imrotate_bin_nocrop(x, r)
+    return greaterthan05(collect(imrotate(x, r; method=BSpline(Constant()))))
+end
+imrotate_bin_clockwise_radians(x, r) = imrotate_bin(x, r)
+imrotate_bin_counterclockwise_radians(x, r) = imrotate_bin(x, -r)
+imrotate_bin_clockwise_degrees(x, r) = imrotate_bin_clockwise_radians(x, deg2rad(r))
+imrotate_bin_counterclockwise_degrees(x, r) = imrotate_bin_counterclockwise_radians(x, deg2rad(r))
+
+
+
+# Functions used for the SD minimization
+"""
+Pad images by zeros based on the size of the larger of the two images.
+"""
+function pad_images(im1, im2)
+    n = max(size(im1)..., size(im2)...)
+    im_padded = [collect(padarray(im, Fill(0, (n, n), (n, n)))) for im in [im1, im2]]
+    return im_padded[1], im_padded[2]
+end
+
+"""
+Calculate the centroid of a binary image. If 'rounded', return the
+nearest integer.
+"""
+function compute_centroid(im; rounded=false)
+    xi = 0
+    yi = 0
+    R = sum(im .> 0)
+    n, m = size(im)
+    for ii in range(1, n)
+        for jj in range(1, m)
+            if im[ii, jj] > 0
+                xi += ii
+                yi += jj
+            end
+        end
+    end
+
+    x0, y0 = sum(xi) / R, sum(yi) / R
+    if rounded
+        return round(Int32, x0), round(Int32, y0)
+    else
+        return x0, y0
+    end
```

Comment:
This pattern is more Julian:
```suggestion
    rounded && return round(Int32, x0), round(Int32, y0)
    return x0, y0
```